File: kanucontrolServer/src/main/java/com/kcserver/dto/simulation/VeranstaltungsInfo.java
package com.kcserver.dto.simulation;

import com.kcserver.enumtype.VeranstaltungTyp;
import lombok.*;

import java.time.LocalDate;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class VeranstaltungsInfo {

    private Long id;
    private String name;
    private LocalDate beginnDatum;
    private LocalDate endeDatum;
    private VeranstaltungTyp typ;
    private long tage;
    private long naechte;

    // Verein
    private boolean vereinKikZertifiziert;

    private String beitragsstrukturName;
    private String unterkunftsartName;
    private String verpflegungsmodellName;
}

File: kanucontrolClient/src/api/types/simulation/VeranstaltungsInfo.ts
// api/types/simulation/VeranstaltungsInfo.ts

import { VeranstaltungTyp } from "@/api/enums/VeranstaltungTyp";

export interface VeranstaltungsInfo {

    id: number;

    name: string;

    beginnDatum: string;

    endeDatum: string;

    typ: VeranstaltungTyp;

    tage: number;

    naechte: number;

    vereinKikZertifiziert: boolean;

    beitragsstrukturId?: number;
    beitragsstrukturName?: string;

    unterkunftsartId?: number;
    unterkunftsartName?: string;

    verpflegungsmodellId?: number;
    verpflegungsmodellName?: string;
}